### DISCUSSION TITLE
Autokey, or something resembling it

### DISCUSSION THREAD

**Discussion by stibinator** [i3/i3#6335] [Q&A] [answered]
I have some autokey bindings for commonly typed phrases, like my email address, work web site etc. and special character I use a lot like ellipsis and em-dash. Saves a lot of typing and futzing around. 

But autokey doesn't seem to work with i3wm, and I can't find any documentation on how to get it to work.

Is there a way of binding phrases to actions? For example my autokey phrase for my work email address is `:@w`, or for an em-dash I type `<scroll lock>---`. These are typed as sequences, not chords. I have too many chords to remember as it is.

So my question is either: how do I get autokey to work with i3wm, or is there a way of binding sequences of character to actions?

**Comment by orestisfl** [comment]
Autokey works with i3 to my best knowledge, what's preventing you from using it?

**Comment by stibinator** [comment]
Hmm, none of my keybindings seem to work. I'm not sure how to start autokey. currently in my config I have
```
exec --no-startup-id autokey-gtk
```
and it doesn't seem to work. If I manually run `autokey-gtk -c` it opens the config window and my phrases work as long as I leave the window open. In the stdout I notice that it  outputs stuff like
```
2024-12-13 10:52:41,036 INFO - autokey.service - Matched Phrase "work_email" having abbreviations "[':@w']" against current input
```
So I tried `systemctl enable autokey.service` but it doesn't find an autokey service.

**Reply by orestisfl** [reply]
It seems that autokey-gtk uses appindicator with no fallback, which is not supported. Try autokey-qt.

**Reply by stibinator** [reply]
That does roughly the same thing. Autokey works only as long as the config window is open which isn't ideal (except it's a qt window not a gtk one). I might ask on the autokey discussions.

**Reply by stibinator** [reply]
Oh, wait. The second time I run it, it doesn't open the window and now my replacements are working. But only if I run it manually at a command prompt. If I put it in the i3 conf file and reload i3 it doesn't work. Maybe I'll try logging out/in

**Reply by stibinator** [reply]
oh, no wait, the autokey preferences window was open on another workspace, when I close it my autokey scripts stop working.

**Comment by stibinator** [comment, marked as answer]
You know what, I reckon I might just switch to espanso.
